crates/types/src/config/common.rs

```diff
@@ -0,0 +1,229 @@
+    #[serde_as(as = "serde_with::DisplayFromStr")]
+    #[cfg_attr(feature = "schemars", schemars(with = "String"))]
+    pub shutdown_timeout: Duration,
+
+    /// # Default async runtime thread pool
+    ///
+    /// Size of the default thread pool used to perform internal tasks.
+    /// If not set, it defaults to the number of CPU cores.
+    pub default_thread_pool_size: Option<usize>,
+
+    /// # Tracing Endpoint
+    ///
+    /// Specify the tracing endpoint to send traces to.
+    /// Traces will be exported using [OTLP gRPC](https://opentelemetry.io/docs/specs/otlp/#otlpgrpc)
+    /// through [opentelemetry_otlp](https://docs.rs/opentelemetry-otlp/0.12.0/opentelemetry_otlp/).
+    ///
+    /// To configure the sampling, please refer to the [opentelemetry autoconfigure docs](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#sampler).
+    pub tracing_endpoint: Option<String>,
+
+    /// # Distributed Tracing JSON Export Path
+    ///
+    /// If set, an exporter will be configured to write traces to files using the Jaeger JSON format.
+    /// Each trace file will start with the `trace` prefix.
+    ///
+    /// If unset, no traces will be written to file.
+    ///
+    /// It can be used to export traces in a structured format without configuring a Jaeger agent.
+    ///
+    /// To inspect the traces, open the Jaeger UI and use the Upload JSON feature to load and inspect them.
+    pub tracing_json_path: Option<String>,
+
+    /// # Tracing Filter
+    ///
+    /// Distributed tracing exporter filter.
+    /// Check the [`RUST_LOG` documentation](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html) for more details how to configure it.
+    pub tracing_filter: String,
+    /// # Logging Filter
+    ///
+    /// Log filter configuration. Can be overridden by the `RUST_LOG` environment variable.
+    /// Check the [`RUST_LOG` documentation](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.EnvFilter.html) for more details how to configure it.
+    pub log_filter: String,
+
+    /// # Logging format
+    ///
+    /// Format to use when logging.
+    pub log_format: LogFormat,
+
+    /// # Disable ANSI in log output
+    ///
+    /// Disable ANSI terminal codes for logs. This is useful when the log collector doesn't support processing ANSI terminal codes.
+    pub log_disable_ansi_codes: bool,
+
+    /// Timeout for idle histograms.
+    ///
+    /// The duration after which a histogram is considered idle and will be removed from
+    /// metric responses to save memory. Unsetting means that histograms will never be removed.
+    #[serde(with = "serde_with::As::<Option<serde_with::DisplayFromStr>>")]
+    #[cfg_attr(feature = "schemars", schemars(with = "Option<String>"))]
+    pub histogram_inactivity_timeout: Option<humantime::Duration>,
+
+    #[serde(flatten)]
+    pub service_client: ServiceClientOptions,
```

Comment:
awesome! i wonder if we need to flatten here though vs having a field name like `client`?

Reply:
I'm pro flattening (3 levels is already a stretch), but we can easily change this before release if we feel that `client` makes ergonomics better. The change won't impact the code structure which is good.

Reply:
ok im relaxed !